mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -713,6 +713,15 @@ private[spark] object RandomForest extends Logging {
         binAggregates.metadata.numSplits(featureIndex) != 0
       }
 
+    // If no valid splits for features, then this split is invalid,
+    // return invalid information gain stats.  Take any split and continue.
+    if (validFeatureSplits.isEmpty) {
+      val parentImpurityCalculator = binAggregates.getImpurityCalculator(0, 0)
+      // Splits is empty, so arbitrarily choose to split on first feature with any threshold
+      return (new ContinuousSplit(0, 0),
```

Comment:
We are using a Continuous split regardless of whether the feature is actually continuous or categorical. It doesn't make a difference in practice, but since it's not much extra work it is probably best to return the appropriate type of split.